### PR TITLE
fix(#478): Update permissions in application

### PR DESCRIPTION
**Pull request by alexstojda**
##### _What existing problem does the pull request solve?_

Updating all app routes with appropriate permissions

##### _Complete these items before requesting reviews_
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [x] Integration Tests
- [x] System Tests
- [x] Documentation

### CODE REVIEW

**Review by KeeganWoods1 — DISMISSED** [review]
Tested with multiple combinations of permission, everything is smooth. Code lgtm!

**Review by WindOfRussia — DISMISSED** [review]
I liked that you split up the resouce routes and made them more obvious. Maybe we can also write Policies to replace/enhance some of the permission checks on routes? That'll bind the permission check to actual model manipulation. Anyhow, I'll approve this, any enhancements can be made it a subsequent PR

**Comment by WindOfRussia** [comment]
Looks like the test got out of date, you'll have to resolve that conflict before merging

**Comment by danseremet** [comment]
I will review this now :+1:

File: routes/web.php [hunk 1]
```diff
@@ -24,46 +24,163 @@
...
+    Route::get('/dashboard', function () {
+        return Inertia\Inertia::render('Dashboard');
+    })->name('dashboard');
+
+    /**
+     * ADMIN
+     */
+    Route::name('admin.')->prefix('admin')->group(function () {
+        //TODO: Admin index route
+
```
Comment (danseremet) [inline routes/web.php:40]: Nice grouping of routes, it is very clean :+1: :)

File: routes/web.php [hunk 2]
```diff
@@ -24,46 +24,163 @@
...
+                ->name('store')
+                ->middleware(['permission:users.create']);
+
+            Route::put('/{user}', [UserController::class, 'update'])
+                ->name('update')
+                ->middleware(['permission:users.update']);
+
+            Route::delete('/{user}', [UserController::class, 'destroy'])
+                ->name('destroy')
+                ->middleware(['permission:users.delete']);
```
Comment (danseremet) [inline routes/web.php:58]: Is this the responsibility of the admin to create users or do the users create themselves?

**Review by danseremet — APPROVED** [review]
I thought this was a big PR, but most lines are indentation changes. The main changes I see you did are in web.php and the necessary controllers to fix the routes and add appropriate permissions.

This PR looks good to me! 
There are some questions below, but I will approve since I don't think they are an issue :)

Just needs to resolve some conflicts and update with main :+1:

**Review by danseremet — CHANGES_REQUESTED** [review]
image

**I just observed that the link when clicking on "My Bookings" is not pointing to the right link, I think it might be due to a missing link change.**

image

File: routes/web.php [hunk 3]
```diff
@@ -24,46 +24,163 @@
...
-      Route::post('app_name', [SettingsController::class, 'storeAppName'])->name('app.name');
+    Route::get('/dashboard', function () {
+        return Inertia\Inertia::render('Dashboard');
+    })->name('dashboard');
+
+    /**
+     * ADMIN
+     */
+    Route::name('admin.')->prefix('admin')->group(function () {
+        //TODO: Admin index route
```
[the remark this answers is not in the document]
Comment (alexstojda) [reply routes/web.php:39]: Yep this is covered by the admin dashboard story

File: routes/web.php (hunk 2 above)
Comment (alexstojda) [reply routes/web.php:58]: As per stakeholder discussion, user creation will be manual

File: app/Http/Controllers/BookingRequestController.php [hunk 4]
```diff
@@ -224,7 +224,7 @@ public function destroy(BookingRequest $booking)
         Reservation::where('booking_request_id', $booking->id)->delete();
         $booking->delete();
 
-        return redirect()->route('bookings.index');
+        return redirect()->back();
```
[the remark this answers is not in the document]
Comment (alexstojda) [reply app/Http/Controllers/BookingRequestController.php:216]: Outside the scope of this PR :)

**Review by KeeganWoods1 — APPROVED** [review]

File: resources/js/Components/Navbar/Navbar.vue [hunk 5]
```diff
@@ -215,7 +234,25 @@ export default {
...
+        return (new RegExp("^" + prefix + ".*", "i")).test(e);
+      })
+    },
     showAdminSubnav() {
+      return this.userHasPermissionWithPrefix("users")
+        || this.userHasPermissionWithPrefix("rooms")
+        || this.userHasPermissionWithPrefix('roles')
+        || this.userHasPermissionWithPrefix('settings');
+    },
+    adminSubnavIsActive() {
```
Comment (EvanDime) [inline resources/js/Components/Navbar/Navbar.vue:255]: I am a big fan of this. Very clean

**Review by Alex-Hayeur — APPROVED** [review]
Very helpful also thanks for the work just generally formatting the code. One step closer to a completed app.

We'll need to add seeded users in our DB for the various roles now I think, so that we can easily check if pages are available or blocked for them. This can be done in a separate task though.

**Pull request merged**